pkg/genericapiserver/genericapiserver.go

```diff
@@ -504,6 +521,102 @@ func (s *GenericAPIServer) init(c *Config) {
 	}
 }
 
+// Exposes the given group versions in API.
+func (s *GenericAPIServer) InstallAPIGroups(groupsInfo []APIGroupInfo) error {
+	for _, apiGroupInfo := range groupsInfo {
+		if err := s.installAPIGroup(&apiGroupInfo); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *GenericAPIServer) installAPIGroup(apiGroupInfo *APIGroupInfo) error {
+	apiPrefix := s.ApiGroupPrefix
+	if apiGroupInfo.IsLegacyGroup {
+		apiPrefix = s.ApiPrefix
+	}
+
+	// Install REST handlers for all the versions in this group.
+	apiVersions := []string{}
+	for _, groupVersion := range apiGroupInfo.GroupMeta.GroupVersions {
+		apiVersions = append(apiVersions, groupVersion.Version)
+
+		apiGroupVersion, err := s.getAPIGroupVersion(apiGroupInfo, groupVersion, apiPrefix)
+		if err != nil {
+			return err
+		}
+		if apiGroupInfo.OptionsExternalVersion != nil {
+			apiGroupVersion.OptionsExternalVersion = apiGroupInfo.OptionsExternalVersion
+		}
+
+		if err := apiGroupVersion.InstallREST(s.HandlerContainer); err != nil {
+			return fmt.Errorf("Unable to setup API %v: %v", apiGroupInfo, err)
+		}
+	}
+	// Install the version handler.
+	if apiGroupInfo.IsLegacyGroup {
```

Comment:
Legacy group has to be parameterized to accept a prefix - because we have two legacy groups, Kube /api and OpenShift /oapi.